app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -0,0 +1,81 @@
+      full_result_response(subscriptions)
+    end
+
+    api :PUT, "/hosts/:id/subscriptions/auto_attach", N_("Trigger an auto-attach of subscriptions")
+    param :id, String, :desc => N_("id of the host"), :required => true
+    def auto_attach
+      sync_task(::Actions::Katello::Host::AutoAttachSubscriptions, @host)
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :GET, "/hosts/:id/subscriptions/events", N_("List Candlepin events for the content host")
+    param :id, String, :desc => N_("UUID of the content host"), :required => true
+    def events
+      respond_for_index :collection => { :results => @candlepin_consumer.events.map { |e| OpenStruct.new(e) } }
+    end
+
+    api :PUT, "/hosts/:host_id/subscriptions/remove_subscriptions"
+    param :subscriptions, Array, :desc => N_("Array of subscriptions to add"), :required => false do
+      param :id, String, :desc => N_("Subscription Pool uuid")
+      param :quantity, Integer, :desc => N_("If specified, remove the first instance of a subscription with matching id and quantity.")
+    end
+    def remove_subscriptions
+      #combine the quantities for duplicate pools into PoolWithQuantities objects
+      pool_id_quantities = params[:subscriptions].inject({}) do |new_hash, subscription|
+        new_hash[subscription['id']] ||= PoolWithQuantities.new(Pool.find(subscription['id']))
+        new_hash[subscription['id']].quantities << subscription['quantity']
+        new_hash
+      end
+
+      @host.subscription_facet.remove_subscriptions(pool_id_quantities.values)
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :PUT, "/hosts/:host_id/subscriptions/add_subscriptions", N_("Add a subscription to a host"), :deprecated => true
+    param :host_id, String, :desc => N_("Id of a host"), :required => false
+    param :subscriptions, Array, :desc => N_("Array of subscriptions to add"), :required => false do
+      param :id, String, :desc => N_("Subscription Pool uuid"), :required => true
+      param :quantity, :number, :desc => N_("Quantity of this subscriptions to add"), :required => true
+    end
+    def add_subscriptions
+      pools_with_quantities = params[:subscriptions].map do |sub_params|
+        PoolWithQuantities.new(Pool.find(sub_params['id']), sub_params['quantity'])
+      end
+
+      sync_task(::Actions::Katello::Host::AttachSubscriptions, @host, pools_with_quantities)
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    private
+
+    def find_host_with_subscriptions
+      @host = resource_finder(::Host.authorized("#{action_permission}_hosts"), params[:host_id])
+      fail _("Host has not been registered with subscription-manager") if @host.subscription_facet.nil?
+      @candlepin_consumer = Katello::Candlepin::Consumer.new(@host.subscription_facet.uuid)
```

Comment:
You've essentially already created this, just privately -- https://github.com/Katello/katello/pull/5695/files#diff-c00617b374a5b937317a274f4ebde4e3R99